### PR TITLE
Add doxygen group, noexcept, and ASSERT tests to Interval

### PR DESCRIPTION
**Pull request by mar-celine**

### CODE REVIEW

File: src/Domain/DomainCreators/Interval.hpp [hunk 1]
```diff
@@ -10,6 +10,8 @@
 #include "Options/Options.hpp"
 
 namespace DomainCreators {
+
+/// ingroup DomainCreatorsGroup
```
Comment (kidder) [inline src/Domain/DomainCreators/Interval.hpp]: \ingroup

**Review by kidder — CHANGES_REQUESTED** [review]
rebase when adding the \

**Review by kidder — CHANGES_REQUESTED** [review]
squash

File: src/Domain/DomainCreators/Interval.cpp [hunk 2]
```diff
@@ -22,14 +22,14 @@ Interval::Interval(
     const typename IsPeriodicIn::type is_periodic_in_x,
     const typename InitialRefinement::type initial_refinement_level_x,
     const typename InitialGridPoints::type initial_number_of_grid_points_in_x,
-    const OptionContext& /*context*/)
```
Comment (nilsdeppe) [inline src/Domain/DomainCreators/Interval.cpp]: While you're here, might as well make these non-const and move (remember clang-tidy)

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
please squash right away

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**